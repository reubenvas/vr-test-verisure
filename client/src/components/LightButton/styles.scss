
@import '../../styles/partials/variables';

.button {
  background-color: $paper;
  color: $charcoal;
}
