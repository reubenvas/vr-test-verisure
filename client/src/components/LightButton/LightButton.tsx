import React from 'react';
import { observer } from 'mobx-react';
import styles from './styles.scss';

import testStore from '../../store/testStore';

type propTypes = {
    label: string;
};

const LightButton = (props: propTypes): React.ReactElement<propTypes> => {
    const { timesClicked } = testStore;
    const { increaseTimesClicked } = testStore;
    return (
        <button onClick={() => increaseTimesClicked(timesClicked + 1)} className={`${styles.button} button is-large`}>{`${props.label} ${timesClicked}`}</button>
    );
};

export default observer(LightButton);
