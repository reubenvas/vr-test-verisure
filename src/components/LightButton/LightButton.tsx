import React from 'react';
import styles from './styles.scss';

type propTypes = {
    label: string;
};

const LightButton = (props: propTypes): React.ReactElement<propTypes> => (
    <button className={styles.button}>{props.label}</button>
);

export default LightButton;
