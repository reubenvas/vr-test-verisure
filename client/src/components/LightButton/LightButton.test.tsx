import React from 'react';
import renderer from 'react-test-renderer';
import LightButton from './LightButton';

describe('DarkButton component', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<LightButton label='Test'/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
