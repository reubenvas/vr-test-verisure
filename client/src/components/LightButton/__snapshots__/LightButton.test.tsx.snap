// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkButton component snapshot renders 1`] = `
<button>
  Test
</button>
`;
